File: src/digitalRoot.java
//https://www.codewars.com/kata/541c8630095125aba6000c00
/*
return the digital root of an integer.
Digital root is the recursive sum of all the digits in a number.
E.g: 132189  -->  1 + 3 + 2 + 1 + 8 + 9 = 24  -->  2 + 4 = 6
 */
public class digitalRoot {
    public static void main(String[] args) {
        System.out.println(digital_root(456));

    }

    public static long digital_root(int n) {
        int returnal = 0;
        String[] x = String.valueOf(n).split("");
        for(String z : x){
            returnal += Integer.parseInt(z);
        }
        if(String.valueOf(returnal).length() > 1){
            return digital_root(returnal);
        }
        return returnal;
    }
}

File: src/Javascript/digitalRoot.js
console.log(digital_root(456))

/*
https://www.codewars.com/kata/541c8630095125aba6000c00/
given a number, return its digital root.
Digital root is the recursive sum of all the digits in a number.
 */
function digital_root(n) {
    const sum = n.toString().split("").reduce((a,b)=>Number.parseInt(a)+Number.parseInt(b)); // splits number into digits, adds digits.
    return sum.length === 1 ? Number.parseInt(sum) : digital_root(sum); // uses recursion to call the function again if length != 1, otherwise sum is returned.
}